File: string/sum-of-digits-of-string-after-convert.cpp
class Solution {
public:
    int getLucky(string s, int k) {
        int sum=0;
        int val;

        for(char i:s){
            val=(i-'a'+1);
            if(val<=9) sum+=val;
            else{
                while(val>0){
                    sum+=val%10;
                    val/=10;
                }
            }
        }
        int n;
        if(k==1) return sum;
        else{
            n=sum;
            sum=0;
            while(--k){
                while(n>0){
                    sum+=n%10;
                    n/=10;
                }
                n=sum;
                sum=0;
            }

        }

        return n;
    }
};

File: 2076-sum-of-digits-of-string-after-convert/sum-of-digits-of-string-after-convert.java
class Solution {
    public int getLucky(String s, int k) {
        
        int sum = 0;

        // Convert characters to numbers and add their digits
        for (char ch : s.toCharArray()) {
            int val = ch - 'a' + 1;

            // If number has 2 digits (like 26)
            sum += val / 10;
            sum += val % 10;
        }

        // Perform remaining transformations
        for (int i = 1; i < k; i++) {
            int temp = 0;

            while (sum > 0) {
                temp += sum % 10;
                sum /= 10;
            }

            sum = temp;
        }

        return sum;
    }
}